### ISSUE TITLE
Resource listing page deletion issue

### ISSUE THREAD

**Issue by jellycow** [laravel-filament/filament#186]
**Describe the bug**
When selecting one or more items from a resource list page and trying to delete them an exception happens.

**To reproduce**
Steps to reproduce the behaviour:
1. Install Filament version >= 1.50
2. Go to a resource listing page
3. Select 1 or more items to delete
4. Click on the delete button
5. Confirm the deletion from the modal
6. The exception happens

**Expected behaviour**
The selected items in the resource list should be deleted from the database and the change reflected in the page

**Screenshots**

Deleting a single item
![Single item delete result](https://user-images.githubusercontent.com/209072/110876213-f7f78600-82ce-11eb-859f-e43de052bae9.jpg)

Deleting multiple items - long numbers in query are fake id and sku values from a product record
![Multi item select](https://user-images.githubusercontent.com/209072/110876324-28d7bb00-82cf-11eb-91ee-0b564ffc9b58.png)

![Multiple item delete result](https://user-images.githubusercontent.com/209072/110876333-2c6b4200-82cf-11eb-87eb-d524ac4808b1.jpg)

**Context**
- Package version: [>= 1.5.0]
- Livewire version: [2.3.0]
- Laravel version: [>= 8.12.0]
- Browser: [Chrome latest]
- Server OS: [OSX 10.11.6]
- PHP version: [7.4.14]
- Database: [MySQL 8.0.23]

**Additional details**
Issue seems to be different for single and multiple selections - with multiple selections, the ids passed into the IN clause in the query do not reflect the record ids that should be selected. Instead they look like they contain the other field values from one of the selected records and not a sequence of integers as expected.  

![database entry](https://user-images.githubusercontent.com/209072/110876641-d0ed8400-82cf-11eb-8ee9-cdfc06c50041.png)

Delete buttons in the item edit pages still work as expected.

I rolled back my repo until I found the version were this began - version 1.4.8 worked as expected and no other packages etc have been updated, only filament version was increased. Happens in all versions from 1.5.0 upwards.

Also still happens if I then update all other packages to the current latest versions.

I have also cleared and rebuilt the database, rerun all migrations, reinstalled all vendor packages and cleared all caches etc

**Comment by danharrin** [comment]
Hey! This is very weird behaviour. Could you send a repo that demonstrates this issue please?

**Comment by jellycow** [comment]
Hi, yup seems weird - so since last night I have totally rebuilt everything and even with a simple clean install of everything I still get the reported issue! I have set up a public repo with a new laravel install & the basic packages I have installed at the moment along with a simple Product model / resource set up for testing (factory set up for Product).

Repo: https://github.com/jellycow/DeletionTest

**Comment by jamessessford** [comment]
Morning @danharrin - I upgraded this morning and ran in to the same issue as described here.

Messing about with the code I found the `deleteSelected` function in the ListRecords class, `Filament\Resources\Pages\ListRecords`, was passing the entire model into the destroy function. I was able to correct this by plucking the id out before casting it to an array. 

https://github.com/laravel-filament/filament/blob/6945428ba2c00e46a129e2eb6bee091e581731ab/src/Resources/Pages/ListRecords.php#L62-L75

Will submit a PR
